README.md

```diff
@@ -40,54 +40,126 @@ git clone https://github.com/fortran-lang/stdlib
 cd stdlib
 ```
 
+
 ### Requirements
 
-The preprocessor ```fypp``` (https://github.com/aradi/fypp) is needed because metaprogramming is used.
-It can be installed using the command line installer ```pip```.
+To build the Fortran standard library you need
+
+- a Fortran 2008 compliant compiler, or better, a Fortran 2018 compliant compiler
+  (GCC Fortran and Intel Fortran compilers are known to work for stdlib)
+- CMake version 3.14 or newer (alternatively Make can be used)
+- a build backend for CMake, like Make or Ninja (the latter is recommended on Windows)
+- the [fypp](https://github.com/aradi/fypp) preprocessor (used as meta-programming tool)
+
+If your system package manager does not provide the required build tools, all build dependencies can be installed with the Python command line installer ``pip``:
+
 ```sh
-pip install fypp
+pip install --user fypp cmake ninja
 ```
 
+Alternatively, you can install the build tools from the conda-forge channel with the conda package manager:
+
+```sh
+conda config --add channels conda-forge
+conda create -n stdlib-tools fypp cmake ninja
+conda activate stdlib-tools
+```
+
+You can install conda using the [miniforge installer](https://github.com/conda-forge/miniforge/releases).
+Also, you can install a Fortran compiler from conda-forge by installing the ``fortran-compiler`` package, which installs GFortran.
+
+
+### Supported Compilers
+
+The following combinations are tested on the default branch of stdlib:
+
+Name | Version | Platform | Architecture
+--- | --- | --- | ---
+GCC Fortran | 7, 8, 9, 10 | Ubuntu 18.04 | x86_64
+GCC Fortran | 7, 8, 9, 10 | MacOS Catalina 10.15 | x86_64
+GCC Fortran | 8 | Windows Server 2019 | x86_64
+GCC Fortran (MSYS) | 10 | Windows Server 2019 | x86_64
+GCC Fortran (MinGW) | 10 | Windows Server 2019 | x86_64, i686
+Intel oneAPI classic | 2021.1 | Ubuntu 20.04 | x86_64
+
+We try to test as many available compilers and platforms as possible.
+A list of tested compilers which are currently not working and the respective issue are listed below.
+
+Name | Version | Platform | Architecture | Status
+--- | --- | --- | --- | ---
+NVIDIA HPC SDK | 20.7, 20.9, 20.11 | Manjaro Linux 20 | x86_64 | [#107](https://github.com/fortran-lang/stdlib/issues/107)
```

Comment:
```suggestion
GCC Fortran | 7 | Ubuntu 18.04 | x86_64
GCC Fortran | 7 | Windows ???  | ????
NVIDIA HPC SDK | 20.7, 20.9, 20.11 | Manjaro Linux 20 | x86_64 | [#107](https://github.com/fortran-lang/stdlib/issues/107)
```

Reply:
Windows 10 on x86_64 I guess, unless you have one of the new aarch64 machines.

Reply:
> Windows 10 on x86_64 I guess, unless you have one of the new aarch64 machines.

I don't know. I never use Windows. My comment was based on #61 (probably Windows 10 x86_64 ?) and #296 (mingw 32-bit ?).

Reply:
> I never use Windows.

Same here.

In case it is a CI workflow than it will run on Windows server 2019, for user reports Windows 10 is the most likely choice. Looking at #296 it is most likely a failure from GCC 7 rather than from the architecture.

Reply:
A quick check on Ubuntu 18.04 with GCC 7.5 shows it is working:
https://github.com/awvwgk/fortran-stdlib/runs/1680206302?check_suite_focus=true

I guess it is an issue with the GCC 7.3 / 7.4 series which are known to be buggy (I had to work around several GCC 7.3 bugs in the past, which are all fixed with GCC 7.5).

Reply:
Nice. I just checked my [bug report ](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=94585): it concerns GCC 7.2.
So if the version is mentioned in the README, then GCC7 could be kept in the first table. However, if we start to mention versions then it could be difficult to keep track of everything. Both solutions are good for me.

Reply:
For the CI tests 7.x is implied with the latest version. I would suggest to be more strict when listing failures and always link with the corresponding issue to make it as clear as possible. In the future this should also include the version of stdlib when this failure was last reported.

Reply:
> For the CI tests 7.x is implied with the latest version. 

Good point!

> I would suggest to be more strict when listing failures and always link with the corresponding issue to make it as clear as possible. In the future this should also include the version of stdlib when this failure was last reported.

Nice suggestion. I agree with it. It is a bit of work, but it will help the user in case of issues!